app/src/main/java/com/example/kursaa/features/episodes/presentation/EpisodeRecyclerViewAdapter.kt

```diff
@@ -0,0 +1,45 @@
+package com.example.kursaa.features.episodes.presentation
+
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.TextView
+import androidx.recyclerview.widget.RecyclerView
+import com.example.kursaa.R
+import com.example.kursaa.core.api.model.EpisodesResponse
+
+class EpisodeRecyclerViewAdapter(private val dataSet: Array<EpisodesResponse>) :
```

Comment:
Ciężko będzie dostarczyć dane przez konstruktor jednocześnie korzystając z DI ze względu na to, że najpierw view model musi te dane dostarczy do widoku. Można to obejść tworząc pusty konstruktor adaptera i dodając metodę, która będzie ustawiać dane. Przykład takiego zastosowania możesz znaleźć tutaj: https://github.com/keresmi/Android-Clean-Architecture-Sample/blob/master/app/src/main/java/pl/akademiaandroida/android_clean_architecture_sample/features/characters/all/presentation/CharacterAdapter.kt. Nie sugeruj się wszystkim co tam znajdziesz, bo jest tam zastosowany view binding, a o tym będzie dopiero w module 11 :)